shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1808,24 +1818,19 @@ - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences {
     _orientationPreferences = new_preferences;
 
     if (@available(iOS 16.0, *)) {
+#if APPLICATION_EXTENSION_API_ONLY
+      UIWindowScene* windowScene = self.viewIfLoaded.window.windowScene;
+      [self performOrientationUpdateOnWindowScene:windowScene];
```

Comment:
I think you could just use this logic instead of anything in the `#else`, I didn't test it though so in my PR I only used it in the `APPLICATION_EXTENSION_API_ONLY` check.